includes/class-wc-stripe-helper.php

```diff
@@ -752,4 +752,85 @@ public static function add_payment_intent_to_order( $payment_intent_id, $order )
 		$order->update_meta_data( '_stripe_intent_id', $payment_intent_id );
 		$order->save();
 	}
+
+	/**
+	 * Adds a source or payment method argument to the request array depending on what sort of
+	 * payment method ID is provided. If ID is neither a source or a payment method ID then nothing
+	 * is added.
+	 *
+	 * @param string $payment_method_id  The payment method ID that should be added to the request array.
+	 * @param array $request             The request representing the arguments that will be sent in the request.
+	 *
+	 * @return array  The updated request array.
+	 */
+	public static function add_payment_method_to_request_array( string $payment_method_id, array $request ): array {
+		if ( 0 === strpos( $payment_method_id, 'src_' ) ) {
+			$request['source'] = $payment_method_id;
+		} elseif ( 0 === strpos( $payment_method_id, 'pm_' ) ) {
+			$request['payment_method'] = $payment_method_id;
+		}
+
+		return $request;
+	}
+
+	/**
+	 * Evaluates whether the object passed to this function is a Stripe Payment Method.
+	 *
+	 * @param stdClass $object  The object that should be evaluated.
+	 * @return bool             Returns true if the object is a Payment Method; false otherwise.
+	 */
+	public static function is_payment_method_object( stdClass $payment_method ): bool {
+		return isset( $payment_method->object ) && 'payment_method' === $payment_method->object;
+	}
+
+	/**
+	 * Evaluates whether a given Stripe Source (or Stripe Payment Method) is reusable.
+	 * Payment Methods are always reusable; Sources are only reusable when the appropriate
+	 * usage metadata is provided.
```

Comment:
Are Payment Methods really always reusable?

Reply:
Admittedly, I'm not sure. I copied that comment from my previous PR but I'll look into it again tomorrow 👍

Reply:
Based on the [payment method usage](https://stripe.com/docs/payments/payment-methods#usage) and [save payment method for later docs](https://stripe.com/docs/payments/save-during-payment?platform=web&client=react#charge-saved-payment-method) there are specific types of payment methods that are always reusable (e.g. cards) and the reusable indicators are all set on the payment intent, not the payment method.

I think this is fine for now. If there's a problem it'll be caught when the payment is attempted (i.e. when we try to confirm the payment intent). If merchants/customers run into issues we can add additional qualifiers, e.g. checking the type of the payment method.

I can't seem to find an exhaustive list of which payment methods are reusable so I don't want to add if-statements to test for that now. Let me know if you disagree, happy to make that change 👍